### PR TITLE
Update Video.js to v7

### LINKED ISSUE #816: Upgrade video.js to version 7
We've had a number of users write in to complain about crashes and out of memory errors in Chrome and Safari. It seems like this is a known issue in the video.js version we use currently. We should upgrade to something more recent, probably from the 7 major release. 

#### upgrade needs to be tested for
- [ ] HLS support, via plugin or otherwise
- [ ] YouTube support
- [ ] other plugins, for switching bitrates
- [ ] custome code for "quad" player

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #816

#### What's this PR do?
Update Video.js to v7 and other related packages.

#### How should this be manually tested?
Upload a video, make sure the video controls work.

### CODE REVIEW

**Comment by mbertrand** [comment]
For uploaded videos, everything seems to work fine, including multi-view videos, and subtitles still display if they exist, but the play button is now at the upper left instead of the center:

Screen Shot 2020-01-13 at 1 40 36 PM

For Youtube videos, it mostly seems to work, but the button at the far right of the toolbar to change quality is no longer visible.  Clicking there will show the menu though:

Screen Shot 2020-01-13 at 2 04 33 PM

To test this out: make a video public, then create a new 'YoutubeVideo' object with a foreign key to that video and set the youtube id to any Youtube video you want, and set status to 'processed'.

One thing I still need to test is non-HLS mp4 videos like the ones imported from TechTV.

**Comment by mbertrand** [comment]
For mp4 videos, getting the same thing as with youtube: the quality selector button on the toolbar is invisible, but clicking on where it should be still brings up the menu.  Might just be a styling issue.

Screen Shot 2020-01-13 at 3 01 16 PM


To test this, create a new Video in django admin, with the following VideoFile objects (s3_object_key, encoding):
 'transcoded/techtv/fcda1c455c8a425c8f6912544ae59bb9/basic.mp4', basic
 'transcoded/techtv/fcda1c455c8a425c8f6912544ae59bb9/hd.mp4', HD
 'transcoded/techtv/fcda1c455c8a425c8f6912544ae59bb9/large.mp4', large
 'transcoded/techtv/fcda1c455c8a425c8f6912544ae59bb9/medium.mp4', medium
 'transcoded/techtv/fcda1c455c8a425c8f6912544ae59bb9/small.mp4', small
'techtv/fcda1c455c8a425c8f6912544ae59bb9/original.flv', original

The bucket for all the above video files should be odl-video-service-transcoded-ci

File: static/js/components/VideoPlayer.js [hunk 1]
```diff
@@ -46,12 +46,21 @@ const makeConfigForVideo = (
...
-    }
+    hlsQualitySelector: {}
+  },
+  controlBar: {
+    children: [
+      "playToggle",
+      "volumePanel",
+      "progressControl",
+      "remainingTimeDisplay",
+      "playbackRateMenuButton",
```
Comment (mbertrand) [inline static/js/components/VideoPlayer.js]: Add `      "subsCapsButton",` for the subtitle control

File: static/js/components/VideoPlayer_test.js [hunk 2]
```diff
@@ -111,18 +111,33 @@ describe("VideoPlayer", () => {
...
+            hlsQualitySelector: {}
+          },
+          youtube:    { ytControls: 2 },
+          controlBar: {
+            children: [
+              "playToggle",
+              "volumePanel",
+              "progressControl",
+              "remainingTimeDisplay",
+              "playbackRateMenuButton",
```
Comment (mbertrand) [inline static/js/components/VideoPlayer_test.js]: Add "subsCapsButton", for the subtitle control

Comment (annagav) [reply static/js/components/VideoPlayer_test.js]: in what order should it appear in the control bar?

Comment (mbertrand) [reply static/js/components/VideoPlayer_test.js]: between playback speed and quality selector:
Screen Shot 2020-01-16 at 3 50 08 PM

**Comment by annagav** [comment]
@pdpinch For youtube videos is it okay if we enable youtube controls, instead using videojs control bar? Take a look at Matt's screenshot above. (also I find youtube controls much more responsive and familiar)

**Comment by pdpinch** [comment]
I think it's OK, as long as the video.js controls aren't interfering with the YouTube controls.

**Comment by annagav** [comment]
@mbertrand I think this ready for your review again.

**Pull request merged**